### DISCUSSION TITLE
🙏 🙏 🙏 getContract({multicall:true}) return contract for multicall

### DISCUSSION THREAD

**Discussion by ChrisChan888** [wevm/wagmi#1901] [Ideas]
Is it possible to implement something like this below:
1.  Can have a strongly typed function hint
2. I can use for loop to create contracts with strongly typed function hint

Function hint:
<img width="520" alt="Screenshot 2023-02-22 at 11 43 43 PM" src="https://user-images.githubusercontent.com/86586482/220676027-8b87260b-2d5f-40e5-b8c2-a28309f97984.png">

```js
const wagmigotchiContract = getContract({
  address: "0xecb504d39723b0be0e3a9aa33d646642d1051ee1",
  abi: wagmigotchiABI,
  multicall: true,
});

const mlootContract = {
  address: "0x1dfe7ca09e99d10835bf73044a23b73fc20623df",
  abi: mlootABI,
  multicall: true,
};

const data = await multicall({
  contracts: [
    wagmigotchiContract.getAlive(),
    wagmigotchiContract.getBoredom(),
    mlootContract.getChest(69),
    mlootContract.getWaist(69),
  ],
});

const contractWithLoop = [];
for (let i = 0; i < lpTokens.length; i++) {
  const lpTokenContract = {
    address: "0x123",
    abi: lpTokenContractABI,
    multicall: true,
  };
  const lpToken = assets[i];
  dataWithLoop.push(
    lpTokenContract.liability(),
    lpTokenContract.cash(),
    lpTokenContract.totalSupply()
  );
}

const dataWithLoop = await multicall({
  contracts: contractWithLoop,
});

```

**Comment by jxom** [comment]
That is not possible. I'd suggest to inline the contract addresses & abis like the example: https://wagmi.sh/core/actions/multicall – where it is still strongly typed (see [description here](https://wagmi.sh/core/actions/multicall#abi)).
